src/App.js

```diff
@@ -1,16 +1,50 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
 
 const App = () => {
+  const [entryData, setEntryData] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
+
+  const updateEntryData = updatedEntry => {
+    const entries = entryData.map(entry => {
+      if (entry.id === updatedEntry.id) {
+        toggleLike(updatedEntry.liked);
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    });
+
+    setEntryData(entries);
+
+  };
+
+  
+  const toggleLike = (liked) => {
```

Comment:
Something to consider: It seems like this function by itself is all that is really needed to keep track of the count of likes in App. It would be possible to just pass this function down to ChatEntry to help determine like count. That way you don't have to also keep track of all of the message objects in state as well and instead just keep "like" state at the ChatEntry level.